### DISCUSSION TITLE
State gets updated, but doesn't trigger a re-render (zustand, LocalStorage)

### DISCUSSION THREAD

**Discussion by augustinanders** [neuefische/questions#577] [Web] [answered]
Currently, I'm trying to implement a select input that allows the user to choose the sorting method for the highscores.

This is my select Input:
```
      <select
        name="sorting-method"
        id="sorting-method"
        onChange={(event) => {
          handleSortingMethod(event);
        }}
      >
        <option value="">--Please choose an option--</option>
        <option value="fails">fails 🤯</option>
        <option value="time">time ⏱️</option>
      </select>
```
And this is the handler function that updates the `highscores` state in the `zustand` store:
```
  const handleSortingMethod = (event) => {
    const currentSortingMethod = event.target.value;

    if (currentSortingMethod === "time") {
      sortByTime();
    } else if (currentSortingMethod === "fails") {
      sortByFailed();
    }
  };
```
My problem is that although the `highscores` state gets updated, the `Highscores`-list does not get rerendered immediately. The list only gets rendered in the desired order when I refresh the site.

How can I immediately rerender the list after updating the state?

For reference, here are the links to the relevant files on the [`userstory_13_change_sorting_method`](https://github.com/augustinanders/2023_03_01_capstone-neuefische_memory/tree/userstory_13_change_sorting_method)-branch of my GitHub repository:

- [Have a look at the Zustand-Store](https://github.com/augustinanders/2023_03_01_capstone-neuefische_memory/blob/userstory_13_change_sorting_method/zustand/useHighscoresStore.js)

- [Have a look at the Highscores-Page](https://github.com/augustinanders/2023_03_01_capstone-neuefische_memory/blob/userstory_13_change_sorting_method/pages/highscores/index.js)

**Comment by doemser** [comment, marked as answer]
Hard to say what's going wrong, but if you just want to sort your state differently, it might be easier to derive the sorting from the state like here:
https://codesandbox.io/s/immutable-meadow-swvqpd?file=/src/App.js

so you don't have to handle the sorting in the store at all

**Reply by augustinanders** [reply]
Thx worked out like this 🚀
